### ISSUE TITLE
Buggy or wrong Cron4j Definition? Impossible to set an "Every X times" crons.

### ISSUE THREAD

**Issue by rkkoszewski** [jmrozanec/cron-utils#33]
Hello, if I'm not wrong the Cron4j definition is wrong for "Every X times" crons.

In Cron4J Manual (http://goo.gl/2CgX2h) to have a task execute every 5 minutes you use:
___*/5 * * * *___
And to have the task execute at XX:05 you use:
___5 * * * *___

Now there seems to be some special case in between which Cron4J supports but its not documented anywhere nor I found it in the manuals and examples of Cron4J and its this one:
___/5 * * * *___

___BUG:___
Cron-Utils interprets "/5 * * * *" as "Every 5 minutes" but in Cron4J it is being interpreted as "once a hour at 05min".

Now this gets even worse with the Cron-Utils mapper. As it maps any "\*/x" and "/x" to "/x". So even if you map from CRON4J to CRON4J it breaks each cron string removing the "\*" having a unexpected result.

Here is a demo code:

```
	public static void main(String[] args) {
		
		// Cron Definitions & Parsers
		CronDefinition unixCronDefinition = CronDefinitionBuilder.instanceDefinitionFor(CronType.UNIX);
		CronDefinition cron4jCronDefinition = CronDefinitionBuilder.instanceDefinitionFor(CronType.CRON4J);
		CronParser unixParser = new CronParser(unixCronDefinition);
		CronParser cron4jParser = new CronParser(unixCronDefinition);

		// From UNIX to CRON4J Mapper
		CronMapper cronMapper = new CronMapper(unixCronDefinition, cron4jCronDefinition);
		
		// ISSUE 1: Every X times BUG for CRON4J
		System.out.println(cronMapper.map(unixParser.parse("*/5 * * * *")).asString()); // WRONG - /5 * * * * (Should be */5 * * * *)
		System.out.println(cronMapper.map(unixParser.parse("/5 * * * *")).asString()); // WRONG - /5 * * * * (Should be */5 * * * *)
		System.out.println(cronMapper.map(unixParser.parse("5 * * * *")).asString()); // OK - 5 * * * *
		
		// Descriptor 
		CronDescriptor descriptor = CronDescriptor.instance(Locale.UK);
		System.out.println(descriptor.describe(cron4jParser.parse("*/5 * * * *"))); // OK - every 5 minutes
		System.out.println(descriptor.describe(cron4jParser.parse("/5 * * * *")));	// WRONG - every 5 minutes (Should be "every hour at minute 5")
		System.out.println(descriptor.describe(cron4jParser.parse("5 * * * *")));	// OK - every hour at minute 5
		
		// ISSUE 2: (Not so important)
		// This is not supported by Cron-Utils
		// In Cron4J Manual http://www.sauronsoftware.it/projects/cron4j/manual.php
		System.out.println(descriptor.describe(cron4jParser.parse("0 5 * * *|8 10 * * *|22 17 * * *")));
		System.out.println(cronMapper.map(unixParser.parse("0 5 * * *|8 10 * * *|22 17 * * *")).asString()); 
		
	}
```

Basically the only thing missing here is that Cron-Utils shouls add a "\*/x" to the evey x times definition. And to parse "/x" the same as "x" and not as "\*/x".

Thanks for the awesome work done so far. Cron-Utils is just awesome useful.

**Comment by jmrozanec** [comment]
@rkkoszewski thank you for reporting this. Looking at cron definitions across implementations, and reading the cron4j documenation as well, I don't think this should be considered cron-utils bug. This may be filed as a cron4j issue, since the behavior you describe does not match a widely accepted cron convention. 

Some examples and definitions I checked:
- [IEEE standard definitions for cron](http://pubs.opengroup.org/onlinepubs/9699919799/)
- [Ubuntu cron documentation](http://manpages.ubuntu.com/manpages/maverick/man5/crontab.5.html)
- [RedHat cron documentation](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/6/html/Deployment_Guide/ch-Automating_System_Tasks.html)
- [Quartz scheduler docs and examples](http://quartz-scheduler.org/documentation/quartz-2.x/tutorials/tutorial-lesson-06)
- [wikipedia page](http://bit.ly/1MtCVOR)

Further comments and opinions are welcome!
